flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/plan/logical/SliceAttachedWindowingStrategy.java

```diff
@@ -46,12 +46,13 @@ public SliceAttachedWindowingStrategy(
             @JsonProperty(value = FIELD_NAME_TIME_ATTRIBUTE_TYPE) LogicalType timeAttributeType,
             @JsonProperty(FIELD_NAME_SLICE_END) int sliceEnd) {
         super(window, timeAttributeType);
+        checkArgument(sliceEnd >= 0);
         this.sliceEnd = sliceEnd;
     }
 
     @Override
     public String toSummaryString(String[] inputFieldNames) {
-        checkArgument(sliceEnd >= 0 && sliceEnd < inputFieldNames.length);
+        checkArgument(sliceEnd < inputFieldNames.length);
```

Comment:
Could we remove this check, because we already check in constructor?

Reply:
No. This is not checked in constructor, the fieldNames is passed in this method.

Reply:
OK